platform/linux-dpdk/odp_crypto.c

```diff
@@ -122,29 +130,24 @@ static int auth_alg_odp_to_rte(odp_auth_alg_t auth_alg,
 	case ODP_AUTH_ALG_MD5_96:
 #endif
 		auth_xform->auth.algo = RTE_CRYPTO_AUTH_MD5_HMAC;
-		auth_xform->auth.digest_length = 12;
```

Comment:
`auth.digest_length` should be set for MD5_96 here, but not for MD5_HMAC.

Reply:
it is set in get_crypto_dev(): auth_xform->auth.digest_length = cap->sym.auth.digest_size.min

Reply:
OK. You always use `param->auth_digest_len` if it is set. This logic is incorrect. It should be used for all cases except MD5_96, SHA256_128, AES128_GCM (where an override should be used).

Reply:
you mean to say for MD5_96, we should ignore the digest_length from application even if it is set ?

Reply:
Yes.

Reply:
in the last review version, you mention that application won't set diget_length for MD5_96 etc....  And also I see that from linux-generic odp doesn't not supporting MD5_96 & SHA256_128 and we shouldn't care about them here I suppose. In face those should be removed in the next version.

Reply:
linux-generic supports them.

Reply:
Ys I see that. Will update to support those algos as well.